src/components/network-chart/index.js

```diff
@@ -40,49 +43,184 @@ const ROW_HEIGHT = 16;
 
 // The SizeProps are injected by the WithSize higher order component.
 type DispatchProps = {|
+  +changeSelectedNetworkMarker: typeof changeSelectedNetworkMarker,
   +changeRightClickedMarker: typeof changeRightClickedMarker,
 |};
 
 type StateProps = {|
+  +selectedNetworkMarkerIndex: MarkerIndex | null,
   +markerIndexes: MarkerIndex[],
   +getMarker: MarkerIndex => Marker,
   +rightClickedMarkerIndex: MarkerIndex | null,
   +disableOverscan: boolean,
   +timeRange: StartEndRange,
   +threadsKey: ThreadsKey,
+  +scrollToSelectionGeneration: number,
 |};
 
 type OwnProps = {| ...SizeProps |};
 
 type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
 class NetworkChartImpl extends React.PureComponent<Props> {
+  _virtualListRef = React.createRef<VirtualList<MarkerIndex>>();
   _memoizedGetSpecialItems = memoize(
-    rightClickedMarkerIndex => {
+    (selectedNetworkMarkerIndex, rightClickedMarkerIndex) => {
+      const specialItems = [undefined, undefined];
+
+      if (selectedNetworkMarkerIndex !== null) {
+        specialItems[0] = selectedNetworkMarkerIndex;
+      }
       if (rightClickedMarkerIndex !== null) {
-        return [rightClickedMarkerIndex];
+        specialItems[1] = rightClickedMarkerIndex;
       }
-      return [];
+      return specialItems;
     },
     { limit: 1 }
   );
 
+  componentDidMount() {
+    this.focus();
+  }
```

Comment:
Please try to have empty lines between the methods, so that it's easier to tell them apart with a quick glance.